multipers/mma_structures.pyx.tp

```diff
@@ -98,11 +99,22 @@ cdef inline get_summand_filtration_values_{{SHORT}}(Summand[{{CTYPE}}] summand):
     """
     Returns a list (over parameter) of the filtrations values of this parameter.
     """
-    cdef vector[Finitely_critical_multi_filtration[{{CTYPE}}]] v = summand.get_birth_list()
+    cdef vector[One_critical_filtration[{{CTYPE}}]] v = summand.get_birth_list()
     births = _vff21cview_{{SHORT}}(v, copy=True)
     v = summand.get_death_list()
     deaths = _vff21cview_{{SHORT}}(v, copy=True)
-    pts = np.concatenate([births,deaths],axis=0)
+
+    if len(births) == 1 and births[0].size == 1 and isinf(births[0][0]):
+        if len(deaths) == 1 and deaths[0].size == 1 and isinf(deaths[0][0]):
+            return []
```

Comment:
Hmm I'm not sure about this return type, I'll check that when I test

Reply:
You return `out = [f[1:]  if f[0] == -np.inf else f for f in out]` in the other cases. Is that not the same type as `[]`?